src/PowerShellEditorServices/Services/PowerShell/Host/InternalHost.cs

```diff
@@ -0,0 +1,778 @@
+            {
+                while (!cancellationScope.CancellationToken.IsCancellationRequested
+                    && _taskQueue.TryTake(out ISynchronousTask task))
+                {
+                    if (task.ExecutionOptions.MustRunInForeground)
+                    {
+                        // If we have a task that is queued, but cannot be run under readline
+                        // we place it back at the front of the queue, and cancel the readline task
+                        _taskQueue.Prepend(task);
+                        _skipNextPrompt = true;
+                        _cancellationContext.CancelIdleParentTask();
+                        break;
+                    }
+
+                    task.ExecuteSynchronously(cancellationScope.CancellationToken);
+                }
+            }
+        }
+
+        private void OnCancelKeyPress(object sender, ConsoleCancelEventArgs args)
+        {
+            _cancellationContext.CancelCurrentTask();
+        }
+
+        private ConsoleKeyInfo ReadKey(bool intercept)
+        {
+            // PSRL doesn't tell us when CtrlC was sent.
+            // So instead we keep track of the last key here.
+            // This isn't functionally required,
+            // but helps us determine when the prompt needs a newline added
+
+            _lastKey = ConsoleProxy.SafeReadKey(intercept, CancellationToken.None);
+            return _lastKey.Value;
+        }
+
+        private bool LastKeyWasCtrlC()
+        {
+            return _lastKey.HasValue
+                && _lastKey.Value.Key == ConsoleKey.C
+                && (_lastKey.Value.Modifiers & ConsoleModifiers.Control) != 0
+                && (_lastKey.Value.Modifiers & ConsoleModifiers.Alt) != 0;
+        }
+
+        private void OnDebuggerStopped(object sender, DebuggerStopEventArgs debuggerStopEventArgs)
+        {
+            DebugContext.SetDebuggerStopped(debuggerStopEventArgs);
+            try
+            {
+                CurrentPowerShell.WaitForRemoteOutputIfNeeded();
+                PushPowerShellAndRunLoop(CreateNestedPowerShell(CurrentRunspace), PowerShellFrameType.Debug | PowerShellFrameType.Nested);
+                CurrentPowerShell.ResumeRemoteOutputIfNeeded();
+            }
+            finally
+            {
+                DebugContext.SetDebuggerResumed();
+            }
+        }
+
+        private void OnBreakpointUpdated(object sender, BreakpointUpdatedEventArgs breakpointUpdatedEventArgs)
+        {
+            DebugContext.HandleBreakpointUpdated(breakpointUpdatedEventArgs);
+        }
+
+        private void OnRunspaceStateChanged(object sender, RunspaceStateEventArgs runspaceStateEventArgs)
+        {
+            if (!_shouldExit && !runspaceStateEventArgs.RunspaceStateInfo.IsUsable())
+            {
+                //PopOrReinitializeRunspaceAsync();
+            }
+        }
+
+        /*
+        private void PopOrReinitializeRunspace()
```

Comment:
I wrote this as a sort of last ditch attempt to stop the extension from falling over when everything goes wrong. The code currently isn't compatible with the new implementation, so we should review it and decide whether it's worth keeping in any way or should just be gotten rid of